File: backend/app/routes/locations.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List, Optional
from .. import models, database, schemas

router = APIRouter()

@router.get("/locations", response_model=List[schemas.LocationResponse])
async def get_locations(
    city: Optional[str] = None,
    category: Optional[str] = None,
    db: Session = Depends(database.get_db)
):
    query = db.query(models.Location)
    if city:
        query = query.filter(models.Location.city.ilike(f"%{city}%"))
    if category:
        query = query.filter(models.Location.category == category)
    return query.all()

@router.post("/locations", response_model=schemas.LocationResponse)
async def create_location(location: schemas.LocationBase, db: Session = Depends(database.get_db)):
    db_location = models.Location(**location.dict())
    db.add(db_location)
    db.commit()
    db.refresh(db_location)
    return db_location

File: frontend/src/shared/api/locations.ts
import { apiClient } from '.';
import type { Location, CreateLocationRequest } from './types/locations';


export const getLocations = async (params?: {
  city?: string;
  category?: string;
}): Promise<Location[]> => {
  const response = await apiClient.get<Location[]>('/locations', { params });
  return response.data;
};


export const createLocation = async (
  data: CreateLocationRequest,
): Promise<Location> => {
  const response = await apiClient.post<Location>('/locations', data);
  return response.data;
};